### ISSUE TITLE
Add more string functions - culture-sensitive functions

### ISSUE THREAD

**Issue by cannorin** [fsprojects/FSharpPlus#165]
### Description

https://github.com/fsprojects/FSharpPlus/issues/158#issuecomment-512721780
> Regarding the functions taking a culture as parameter, I propose also to discuss them in a different issue, because I think they need more in depth study as I can't find a good proposal at the moment. So let's move them to another issue in order to speed up this one.

"culture-sensitive functions" are functions taking [`CultureInfo`](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo) as s parameter. 

### Candidates

* `Compare`
* `ToLower / ToUpper`
* `StartsWith / EndsWith`
* (add more if something is missing)

### Ideas

* adding suffix like `WithCulture` 
* using functor(a la OCaml)-like dependency injection
* to be discussed

**Comment by wallymathieu** [comment]
How would functor logic a la Ocaml like dependency injection look?

**Comment by cannorin** [comment]
`String.Cultural(culture).toLower`, I guess?

**Comment by gusty** [comment]
Here's an example:

    type CString (ci: System.Globalization.CultureInfo) =
        member __.ToUpper (s: string)  = s.ToUpper (ci)
        member __.ToLower (s: string)  = s.ToLower (ci)


    // usage

    let String = CString (System.Globalization.CultureInfo.CreateSpecificCulture "PT")

    let r = String.ToLower "Sample String"

So, the `let String = CString (System.Globalization.CultureInfo.CreateSpecificCulture "PT")` is like opening a module (actually more like aliasing it) with a specific parameter.

**Comment by wallymathieu** [comment]
Yes, makes sense. Feels like a thing that you should be able to do perhaps nicer in f#, i.e.:

```fsharp
module String = CString.Make (System.Globalization.CultureInfo.CreateSpecificCulture "PT")
```
